ompi/request/request.h

```diff
@@ -428,31 +429,59 @@ static inline void ompi_request_wait_completion(ompi_request_t *req)
  *  this function, or the synchronization primitive might not be correctly
  *  triggered.
  */
+
 static inline int ompi_request_complete(ompi_request_t* request, bool with_signal)
 {
     int rc = 0;
-
-    if( NULL != request->req_complete_cb) {
-        rc = request->req_complete_cb( request );
+    
+    OPAL_THREAD_LOCK(request->req_lock);
```

Comment:
This lock is going to cost multithreaded a lot of time. Is there a way to turn it on only when your module is loaded?